internal/directives/promotions.go

```diff
@@ -96,16 +101,29 @@ type PromotionStep struct {
 	Config []byte
 }
 
-// GetMaxAttempts returns the maximum number of attempts that can be made to
-// execute the step using the provided runner. If the runner is a
-// RetryableStepRunner, the value of its retry configuration is used as the
-// maximum default. Otherwise, the default is 1.
-func (s *PromotionStep) GetMaxAttempts(runner any) int64 {
-	fallback := int64(1)
+// GetTimeout returns the maximum interval the provided runner may spend
+// attempting to execute the step before retries are abandoned and the entire
+// Promotion is marked as failed. If the runner is a RetryableStepRunner, its
+// timeout is used as the default. Otherwise, the default is 0 (no limit).
+func (s *PromotionStep) GetTimeout(runner any) *time.Duration {
+	fallback := ptr.To(time.Duration(0))
 	if retryCfg, isRetryable := runner.(RetryableStepRunner); isRetryable {
-		fallback = retryCfg.DefaultAttempts()
+		fallback = retryCfg.DefaultTimeout()
 	}
-	return s.Retry.GetAttempts(fallback)
+	return s.Retry.GetTimeout(fallback)
+}
+
+// GetErrorThreshold returns the number of consecutive times the provided runner
+// must fail to execute the step (for any reason) before retries are abandoned
+// and the entire Promotion is marked as failed. If the runner is a
+// RetryableStepRunner, its threshold is used as the default. Otherwise, the
+// default is 1.
+func (s *PromotionStep) GetErrorThreshold(runner any) uint32 {
+	const fallback = uint32(1)
```

Comment:
This can't be const. It potentially takes on a different value a few lines below.

```suggestion
	fallback := uint32(1)
```